--- dnf5-5.4.6.0/libdnf5/rpm/arch_private.hpp.orig	2026-09-24 20:44:28.000000000 +0200
+++ dnf5-5.4.6.0/libdnf5/rpm/arch_private.hpp	2026-09-27 12:05:49.643888455 +0200
@@ -66,6 +66,7 @@ static const struct {
     {"x86_64", {"x86_64", "amd64", "ia32e", nullptr}},
     {"loongarch64", {"loongarch64", nullptr}},
     {"e2k", {"e2k", "e2kv4", "e2kv5", "e2v6"}},
+    {"x32", {"x32", nullptr}},
     {nullptr, {nullptr}}};
 
 }  // namespace libdnf5::rpm
